rousseau_vote/lib/src/widgets/vote/vote_bar.dart

```diff
@@ -0,0 +1,144 @@
+        getVoteButton(context)
+      ],
+    );
+  }
+  
+  Widget getVoteButton(BuildContext context) {
+    if (!_poll.isOpen() || _poll.alreadyVoted) {
+      return Container();
+    }
+    return SizedBox(
+      width: double.infinity,
+      height: 60,
+      child: RaisedButton(
+        child: Text(
+          RousseauLocalizations.getText(context, 'vote-button').toUpperCase(),
+          style: TextStyle(color: Colors.white, fontWeight: FontWeight.bold, fontSize: 18),
+        ),
+        color: _selected.isEmpty ? DISABLED_GREY : PRIMARY_RED,
+        onPressed: () => _selected.isEmpty ? showMessage(context) : showAction(context),
+      ),
+    );
+  }
+
+  Widget getPreferencesBar(BuildContext context) {
+    if(_poll.alreadyVoted || _poll.calculatePollStatus() == PollStatus.CLOSED){
+     return Container();
+    }
+
+    final int remainingPreferences = _poll.maxSelectableOptionsNumber - _selected.length;
+    if(remainingPreferences == 0){
+      return Text(
+        RousseauLocalizations.getText(context, 'vote-preferences-done'),
+        style: const TextStyle(fontSize: 18),
+      );
+    }
+    final String text1 = RousseauLocalizations.getText(context, 'vote-preferences-1');
+    String text2 = '';
+    if (remainingPreferences == 1) {
+      text2 += RousseauLocalizations.getText(context, 'vote-preferences-2s');
+    } else {
+      text2 += RousseauLocalizations.getText(context, 'vote-preferences-2p');
+    }
+    return Text.rich(
+      TextSpan(
+        text: text1,
+        style: const TextStyle(fontSize: 18),
+        children:<TextSpan>[
+          TextSpan(
+            text: remainingPreferences.toString(),
+            style: TextStyle(
+              fontSize: 28,
+              fontWeight: FontWeight.w600
+            )
+          ),
+          TextSpan(
+            text: text2
+          )
+        ],
+      ),
+      textAlign: TextAlign.center,
+    );
+  }
+
+  void showMessage(BuildContext context) {
+    Scaffold.of(context).showSnackBar(
+        SnackBar(
+          content: Text(RousseauLocalizations.getText(context, 'vote-pick-one')),
+          action: SnackBarAction(
+            label: RousseauLocalizations.getText(context, 'close'),
+            onPressed: () => Scaffold.of(context).hideCurrentSnackBar()
+          ),
+        )
+      );
+  }
+
+  void showAction(BuildContext context) {
+    showDialog<AlertDialog>(
+      context: context,
+      builder: (BuildContext context) {
+        return VoteDialog(
+          _selected, 
+          _poll.slug,
+          showError,
+          showDone
+        );
+      }
+    );
+  }
+
+  void showDone(BuildContext context) {
+    Navigator.of(context).pop();
+    showDialog<AlertDialog>(
+      context: context,
+      builder: (BuildContext context) {
+        return DoneDialog(_endAction);
+      }
+    );
+  }
+
+  void showError(BuildContext context) {
+    Navigator.of(context).pop();
+    showDialog<AlertDialog>(
```

Comment:
We should refactor this and pull it to ui_util (there might be something similar there). So, it can be reused all over the app.

Reply:
ok, done.